src/loci/formats/S3FileSystemStore.java

```diff
@@ -0,0 +1,293 @@
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ * #L%
+ */
+
+import com.bc.zarr.ZarrConstants;
+import com.bc.zarr.ZarrUtils;
+import com.bc.zarr.storage.Store;
+
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.file.FileSystem;
+import java.nio.file.Files;
+import java.nio.file.LinkOption;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class S3FileSystemStore implements Store {
+
+    private Path root;
+    AmazonS3 client;
+    public static final String ENDPPOINT_PROTOCOL= "https://";
+    protected static final Logger LOGGER =
+        LoggerFactory.getLogger(S3FileSystemStore.class);
+
+    public S3FileSystemStore(String path, FileSystem fileSystem) {
+        if (fileSystem == null) {
+            root = Paths.get(path);
+        } else {
+            root = fileSystem.getPath(path);
+        }
+        setupClient();
+    }
+    
+    public void updateRoot(String path) {
+      root = Paths.get(path);
+    }
+    
+    public String getRoot() {
+      return root.toString();
+    }
+
+    private void setupClient() {
+      String[] pathSplit = root.toString().split(File.separator);
+      String endpoint = "ENDPPOINT_PROTOCOL" + pathSplit[1] + File.separator;
+      try {   
+        client = AmazonS3ClientBuilder.standard()
+          .withEndpointConfiguration(new AwsClientBuilder.EndpointConfiguration(endpoint, "auto"))
+          .withPathStyleAccessEnabled(true)
+          .withCredentials(new AWSStaticCredentialsProvider(new AnonymousAWSCredentials())).build();
+      } catch (Exception e) {
+        LOGGER.info("Exception caught while constructing S3 client");
+        e.printStackTrace();
+      } 
+      
+    }
+    
+    public void close() {
+      if (client != null) {
+        client.shutdown();
+      }
+    }
+
+    public S3FileSystemStore(Path rootPath) {
+        root = rootPath;
+        setupClient();
```

Comment:
In that case, does `setupClient` need to be invoked at all?

Reply:
Yeah setupClient is needed for any of the entry points when constructing the S3FileSystemStore